src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -196,24 +222,59 @@ def iot_dps_access_policy_delete(cmd, client, dps_name, resource_group_name, acc
     if no_wait:
         return client.iot_dps_resource.begin_create_or_update(resource_group_name, dps_name, dps_description)
     LongRunningOperation(cmd.cli_ctx)(client.iot_dps_resource.begin_create_or_update(resource_group_name, dps_name, dps_description))
-    return iot_dps_access_policy_list(client, dps_name, resource_group_name)
+    return iot_dps_policy_list(client, dps_name, resource_group_name)
 
 
 # DPS linked hub methods
-def iot_dps_linked_hub_list(client, dps_name, resource_group_name):
+def iot_dps_linked_hub_list(client, dps_name, resource_group_name=None):
     dps = iot_dps_get(client, dps_name, resource_group_name)
     return dps.properties.iot_hubs
 
 
-def iot_dps_linked_hub_get(client, dps_name, resource_group_name, linked_hub):
+def iot_dps_linked_hub_get(cmd, client, dps_name, linked_hub, resource_group_name=None):
+    if '.' not in linked_hub:
+        hub_client = iot_hub_service_factory(cmd.cli_ctx)
+        linked_hub = _get_iot_hub_hostname(hub_client, linked_hub)
+
     dps = iot_dps_get(client, dps_name, resource_group_name)
     for hub in dps.properties.iot_hubs:
         if hub.name == linked_hub:
             return hub
     raise CLIError("Linked hub '{0}' does not exist. Use 'iot dps linked-hub show to see all linked hubs.".format(linked_hub))
 
 
-def iot_dps_linked_hub_create(cmd, client, dps_name, resource_group_name, connection_string, location, apply_allocation_policy=None, allocation_weight=None, no_wait=False):
+def iot_dps_linked_hub_create(
+    cmd,
+    client,
+    dps_name,
+    hub_name=None,
+    hub_resource_group_name=None,
+    connection_string=None,
+    location=None,
+    resource_group_name=None,
+    apply_allocation_policy=None,
+    allocation_weight=None,
+    no_wait=False
+):
+    if not connection_string and not hub_name:
+        raise CLIError("Please provide an IoT Hub connection string or the name of the IoT Hub.")
+    elif not connection_string:
+        # Get the connection string for the hub
+        hub_client = iot_hub_service_factory(cmd.cli_ctx)
+        connection_string = iot_hub_show_connection_string(
+            hub_client, hub_name=hub_name, resource_group_name=hub_resource_group_name
+        )['connectionString']
+
+    if not location:
+        if not hub_name:
+            # Ensure there is a hub name
+            name_start = connection_string.find("HostName=") + len("HostName=")
+            hub_name = connection_string[name_start : connection_string.find(".", name_start)]
+
+        hub_client = iot_hub_service_factory(cmd.cli_ctx)
+        location = iot_hub_get(cmd, hub_client, hub_name=hub_name, resource_group_name=hub_resource_group_name).location
+
+    resource_group_name = _ensure_dps_resource_group_name(client, resource_group_name, dps_name)
```

Comment:
Is the below Hack still required where the 2017-11-15 is being used?

Reply:
linked-hub create still works without the api change hack